### PR TITLE
Add /boot options to ANSSI kickstarts and remediation for mount_option_nodev_nonroot_local_partitions

### PR DESCRIPTION
**Pull request by mildas**
#### Description:
Add `nodev` option to `/boot` partition for ANSSI kickstarts with intermediary level or higher.

#### Rationale:
ANSSI profiles have [rule](https://github.com/ComplianceAsCode/content/blob/master/controls/anssi.yml#L177) that requires `nodev` option on all non-root partitions. The `/boot` partition doesn't have the option and because of that the rules fails after installation.

### CODE REVIEW

File: rhel7/kickstart/ssg-rhel7-anssi_nt28_enhanced-ks.cfg [hunk 1]
```diff
@@ -99,7 +99,7 @@ zerombr
 clearpart --linux --initlabel
 
 # Create primary system partitions (required for installs)
-part /boot --fstype=xfs --size=512
+part /boot --fstype=xfs --size=512 --fsoptions="nodev"
```
Comment (yuumasato) [inline rhel7/kickstart/ssg-rhel7-anssi_nt28_enhanced-ks.cfg:102]: What do you think of setting all recommended options for `/boot`? Which are `nosuid,nodev,noexec`.
All other configured partitions have them all.

**Review by yuumasato — COMMENTED** [review]
Adding `nodev` option to the kickstart is just part of the solution.
The `nodev` rule in the profiles (`mount_option_nodev_nonroot_local_partitions`) doesn't have a remediation (I'm not saying this should be done in this PR, :P).

File: rhel7/kickstart/ssg-rhel7-anssi_nt28_enhanced-ks.cfg (hunk 1 above)
Comment (mildas) [reply rhel7/kickstart/ssg-rhel7-anssi_nt28_enhanced-ks.cfg:102]: Sure, I will add them.

**Comment by mildas** [comment]
@yuumasato I've added the remediation and test scenarios. Test scenarios are passing but I'm not sure if this is a proper way how to test partition options.

**Comment by yuumasato** [comment]
@mildas How did you the The test suite? It doesn't work for me, :thinking: 
```
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/wsato/git/content/tests/logs/rule-custom-2021-02-09-1322/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_mount_option_nodev_nonroot_local_partitions                                                        
ERROR - Failed to execute 'cd /root/ssgts/mount_option_nodev_nonroot_local_partitions; SHARED=/root/ssgts/shared bash -x no_nonroot_partitio
ns.pass.sh' on root@192.168.122.14: Command '('ssh', '-o', 'StrictHostKeyChecking=no', '-o', 'UserKnownHostsFile=/dev/null', 'root@192.168.1
22.14', 'cd /root/ssgts/mount_option_nodev_nonroot_local_partitions; SHARED=/root/ssgts/shared bash -x no_nonroot_partitions.pass.sh')' retu
rned non-zero exit status 32.
ERROR - Terminating due to error: Failed to execute 'cd /root/ssgts/mount_option_nodev_nonroot_local_partitions; SHARED=/root/ssgts/shared b
ash -x no_nonroot_partitions.pass.sh' on root@192.168.122.14.
```

File: linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/tests/correct.pass.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+. $SHARED/partition.sh
+
+umount --all
+rm /etc/fstab
+PARTITION="/dev/new_partition2"; create_partition
```
Comment (yuumasato) [inline linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/tests/correct.pass.sh:7]: Is this line a copy paste leftover?

**Comment by mildas** [comment]
@yuumasato
```
$ python3 tests/test_suite.py rule --libvirt qemu:///system test-suite-rhel8 build/ssg-rhel8-ds.xml mount_option_nodev_nonroot_local_partitions
```

**Review by yuumasato — COMMENTED** [review]
The test setup looks ok.

File: linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/tests/correct.pass.sh (hunk 2 above)
Comment (mildas) [reply linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/tests/correct.pass.sh]: Yea :D and it's gone

**Comment by yuumasato** [comment]
@mildas I can't run these test scenarios, both rhel8 and rhel7 VMs fail with error described above.
Tests for other rules run fine.

File: linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/tests/correct.pass.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+. $SHARED/partition.sh
+
+umount --all
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/tests/correct.pass.sh:5]: What is the purpose of this line? Currently, it renders system kinda unusable and tests can't continue.

Comment (mildas) [reply linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/tests/correct.pass.sh:5]: To ensure that everything is unmounted except partitions that are specified below. Idea is to have "clean" environment for testing and not to specify them individually.

Comment (yuumasato) [reply linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/tests/correct.pass.sh:5]: When I remove `umount --all` the tests are executed, although they fail.

**Comment by mildas** [comment]
@yuumasato @vojtapolasek Updated.
I had to update remediation to use `findmnt` instead `egrep` from `/etc/fstab` because some partitions shows as `UUID=...` in `/etc/fstab` instead of its device name (`/dev/..`). And changed approach in test scenarios - first prepare correct environment for test (`nodev` to all and remount with the option) and then perform actual testing phase.

**Comment by mildas** [comment]
Preparation part for actual test is a bit complicated in my opinion. But I don't think it's worthy to create shared functions because there is no other rule that could use that.

**Comment by vojtapolasek** [comment]
Now tests are passing for me. Good job. Are you planing to create Ansible remediation as well?

**Review by yuumasato — APPROVED** [review]
LGTM, all test scenarios pass, :)

**Comment by yuumasato** [comment]
@vojtapolasek Thank you for the hint about `ansible_facts`, it really simplified the remediation.

**Comment by mildas** [comment]
Ansible remediation looks great, all tests are passing. Thanks @yuumasato and @vojtapolasek

**Pull request merged**